src/k2/persistence/logStream/LogStream.cpp

```diff
@@ -0,0 +1,547 @@
+    Payload temp_payload(Payload::DefaultAllocator);
+    temp_payload.write(name);
+    temp_payload.write(sealed_offset);
+    temp_payload.write(std::move(new_plogId));
+    return append(std::move(temp_payload))
+    .then([this] (auto&& response){
+        auto& [status, plogId, appended_offset] = response;
+        K2LOG_D(log::lgbase, "{}, {}", plogId, appended_offset);
+        return seastar::make_ready_future<Status>(std::move(status));
+    });
+}
+
+
+seastar::future<Status> 
+MetadataMgr::replay(String cpo_url, String partitionName, String persistenceClusterName){
+    return init(cpo_url, partitionName, persistenceClusterName, true).
+    then([&] (){
+        return _cpo.GetMetadata(Deadline<>(_cpo_timeout()), _partitionName);
+    })
+    .then([&] (auto&& response){
+        auto& [status, resp] = response;
+        if (!status.is2xxOK()) {
+            throw k2::dto::MetadataGetError("unable to get metadata from CPO");
+        }
+        std::vector<dto::MetadataRecord> records = std::move(resp.records);
+        uint32_t read_size = 0;
+        for (auto& record: records){
+            read_size += record.sealed_offset;
+        }
+        Payload read_payload;
+        return seastar::do_with(std::move(records), std::move(read_size), [&] (auto& records, auto& read_size){
+            return get_plog_status(records.back().plogId)
+            .then([&] (auto&& response){
+                auto& [status, resp] = response;
+                if (!status.is2xxOK()) {
+                    throw k2::dto::PlogGetStatusError("unable to obtain the information of a plog");
+                }
+
+                records.back().sealed_offset = std::get<0>(resp);
+                read_size += std::get<0>(resp);
+                return reload(records);
+            })
+            .then([&] (auto&& response){
+                auto& status = response;
+                if (!status.is2xxOK()) {
+                    throw k2::dto::LogStreamBaseReload("unable to reload the metadata");
+                }
+                return read(records[0].plogId, 0, read_size);
+            })
+            // read with continuation
+            .then([&] (auto&& response){
+                uint32_t request_size = read_size;
+                auto& [status, continuation_token, payload] = response;
+                if (!status.is2xxOK()) {
+                    throw k2::dto::LogStreamBaseReload("unable to read the metadata");
+                }
+                request_size -= payload.getSize();
+                Payload read_payload;
+                for (auto& b: payload.shareAll().release()) {
+                    read_payload.appendBinary(std::move(b));
+                }
+                
+                return seastar::do_with(std::move(read_payload), std::move(request_size), std::move(continuation_token), [&] (auto& read_payload, auto& request_size, auto& continuation_token){
```

Comment:
Please add a bit more comments to explain what these do

Reply:
added more comments here